src/Implement/RequestParser.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "RequestParser.h"
+
+RequestParser* RequestParser::only_request_parser = nullptr;
+
+RequestParser* RequestParser::GetInstance() {
+  if (only_request_parser == nullptr) {
+    only_request_parser = new RequestParser();
+  }
+  return only_request_parser;
+}
+
+// normally http content which will be received in string type
+/*
+POST /?id=1 HTTP/1.1
+Host: echo.paw.cloud
+Content-Type: application/json; charset=utf-8
+User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:53.0)
+Gecko/20100101 Firefox/53.0 Connection: close Content-Length: 136
+
+{
+  "status": "ok",
+  "extended": true,
+  "results": [
+    {"value": 0, "type": "int64"},
+    {"value": 1.0e+3, "type": "decimal"}
+  ]
+}
+*/
+OperateData* RequestParser::ParseData(const std::string& request) {
+  OperateData* new_data = new OperateData();
+  new_data->header["method"] = request.substr(0, request.find(" "));
+  size_t url_key_start = request.find(" ") + 2;
+  size_t url_key_end = request.find(" ", request.find(" ") + 1);
+  new_data->header["url"] =
+      request.substr(url_key_start, url_key_end - url_key_start);
+  for (int i = 0; i < new_data->header["url"].length(); i++) {
```

Comment:
Maybe add comments on what each segment is doing?

Reply:
Done